shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -749,6 +774,513 @@ export class CookieManagerTests extends AITestClass {
+                Assert.equal(newValue1, manager.get(newKey1), "Should still return first value from storage");
+                
+                manager.setEnabled(true);
+                Assert.equal(newValue2 + "; path=/", this._testCookies[newKey2], "Second cookie should be flushed");
+                Assert.equal(newValue1 + "; path=/", this._testCookies[newKey1], "First cookie should still be in storage");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Write cookies after being enabled - unload clears cache",
+            test: () => {
+                let manager = createCookieMgr(this._config);
+
+                // Start with cookies disabled and set some cached cookies
+                manager.setEnabled(false);
+                let newKey = "test." + newId();
+                let newValue = newId();
+                manager.set(newKey, newValue);
+                Assert.equal(newValue, manager.get(newKey), "Should return cached value");
+
+                // Unload the manager
+                manager.unload && manager.unload();
+
+                // Enable cookies - nothing should be flushed since cache was cleared
+                manager.setEnabled(true);
+                Assert.equal(undefined, this._testCookies[newKey], "Cookie should not be in storage after unload");
+                Assert.equal("", manager.get(newKey), "Should return empty string after unload");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Dynamic config change to enable cookies should flush cached values",
+            test: () => {
+                // Create a dynamic config that can be updated
+                let configValues = {
+                    cookieCfg: objExtend({}, this._cookieMgrCfg, {
+                        enabled: false
+                    })
+                };
+                let manager = createCookieMgr(configValues);
+
+                let newKey = "test." + newId();
+                let newValue = newId();
+
+                // Set a cookie while disabled - should be cached
+                let result = manager.set(newKey, newValue);
+                Assert.equal(true, result, "Set should return true even when disabled (cached)");
+                Assert.equal(undefined, this._testCookies[newKey], "Cookie should not be in storage when disabled");
+
+                // Get should return the cached value
+                Assert.equal(newValue, manager.get(newKey), "Should return cached value when disabled");
+
+                // Enable cookies via dynamic config change - should flush cached values
+                configValues.cookieCfg.enabled = true;
+
+                // Verify the cookie was flushed to actual storage
+                Assert.equal(newValue + "; path=/", this._testCookies[newKey], "Cookie should be flushed to actual storage via config change");
```

Comment:
You need to add a clock.tick(1) after changing the state to allow the dynamic configurations changes to take affect